includes/Assets.php

```diff
@@ -144,6 +152,38 @@ public function register_script_asset( string $script_handle, array $script_depe
 		);
 
 		wp_set_script_translations( $script_handle, 'web-stories' );
+
+		// Register every dynamically imported chunk as a script, just so
+		// that we can print their translations whenever the main script is enqueued.
+		// The actual enqueueing of these chunks is done by the main script via dynamic imports.
+		foreach ( $asset['chunks'] as $dynamic_chunk ) {
+			$this->register_script(
+				$dynamic_chunk,
+				$base_script_path . $dynamic_chunk . '.js',
+				[],
+				WEBSTORIES_VERSION,
+				$in_footer
+			);
+
+			wp_set_script_translations( $dynamic_chunk, 'web-stories' );
+
+			$json_translations = load_script_textdomain( $dynamic_chunk, 'web-stories', null );
+
+			if ( ! $json_translations ) {
+				// Register empty locale data object to ensure the domain still exists.
+				$json_translations = '{ "locale_data": { "messages": { "": {} } } }';
+			}
+
+			$output = <<<JS
+( function( domain, translations ) {
+	var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
+	localeData[""].domain = domain;
+	wp.i18n.setLocaleData( localeData, domain );
+} )( "web-stories", {$json_translations} );
+JS;
```

Comment:
This worked in my testing. 

```suggestion
			$output = wp_scripts()->print_translations( $dynamic_chunk, false);
```